frontend/src/components/auth/student-login/StudentLogin.tsx

```diff
@@ -12,33 +20,72 @@ const StudentLogin = (): React.ReactElement => {
   const subtitle = "Please enter your classroom's access code";
   const image = STUDENT_SIGNUP_IMAGE;
 
-  const handleComplete = (code: string) => {};
+  const [error, setError] = useState("");
+  const [success, setSuccess] = useState(false);
+  const [pin, setPin] = useState("");
+  const [pinBorderColor, setPinBorderColor] = useState("grey.100");
+  const [checkPin] = useLazyQuery(GET_TEST_SESSION_BY_ACCESS_CODE, {
+    onCompleted: () => {
+      setPinBorderColor("green.200 !important");
+      setError("");
+      setSuccess(true);
+    },
+    onError: async () => {
+      setPinBorderColor("red.200 !important");
+      setError("Please ensure input is correct");
+      setSuccess(false);
+    },
+  });
+
+  const handleComplete = (code: string) => {
+    checkPin({
+      variables: {
+        accessCode: code,
+      },
+    });
+  };
+
+  const resetPin = () => {
+    if (error) {
+      setPin("");
+      setError("");
+      setPinBorderColor("grey.100");
+    }
+  };
 
   const form = (
     <>
+      {success && (
+        <Text color="green.300" textStyle="smallerParagraph">
+          Entered successfully
+        </Text>
+      )}
       <HStack>
         <PinInput
           autoFocus
           focusBorderColor="grey.200"
+          onChange={(value: string) => setPin(value)}
           onComplete={handleComplete}
           placeholder=""
           size="lg"
+          value={pin}
           variant="filled"
         >
           {[...Array(6)].map((i) => (
             <PinInputField
               key={i}
               _focus={{ backgroundColor: "grey.100" }}
               _hover={{ backgroundColor: "grey.100" }}
-              _invalid={{ borderColor: "red.200" }}
-              _valid={{ backgroundColor: "green.200" }}
               backgroundColor="grey.100"
+              borderColor={pinBorderColor}
```

Comment:
nit: I think instead of using `pinBorderColor` as a separate state, since it entirely depends on `success` and `error` values, you can just create a function above which queries for the value of `borderColor`! then can avoid the use of the `!important` override

i.e define a `getPinBorderColor()`

Reply:
oooo okay makes sense! i think ideally, after the pin completes, none of the fields should be focused? then shouldn't need to override! but i think if this is difficult, we can just use the `!important` in the function.

re: where to call the function - you can set `borderColor={getPinBorderColor()}` as a `PinInputField` attribute~! & then it will dynamically assign `borderColor`. i think even with the `!important` attribute, we want to try keep data stored in states, not the attributes derived from that data! lmk if you have thoughts

Reply:
also, small bug i just noticed is that i think there might be some issue w/ the focusing? 🤔 i noticed that it doesnt input numbers where my cursor is, i.e if i set the cursor to the last pin input, it still fills the first non-empty one. I think this is because the pin fields should be setting input elements individually! i.e the third `PinInputField` should update the value of pin[3], not the whole string.

https://user-images.githubusercontent.com/41273486/225983102-bb3f2d9b-a8e6-43b6-a73d-0bf27f106c08.mov

Reply:
i'm still seeing the bug described above? let me know if you're having trouble reproducing it!

Reply:
Oh sorry I misunderstood ur comment. I think that's a problem with mui's pin input, since there is a wrapper for all the pininputfields called pininput which takes in the whole entire value. But to avoid this problem, I can try to disable all the other inputs before the user inputs? would that work?

Reply:
sounds good! no worries at all. we can work through it together if youd like, just lmk! (:

Reply:
should be fixed now!